The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5866781114/job/15906253263) [(plain)](https://github.com/rust-lang/rust/commit/ff6710fb904fb55d5bc02ab979022db3aaf0347e/checks/15906253263/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_eca1a000-a0f1-4c1f-83a9-465be9ba740a
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=khei4/trailing_zero_codegen
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_eca1a000-a0f1-4c1f-83a9-465be9ba740a
GITHUB_REF=refs/pull/114850/merge
GITHUB_REF_NAME=114850/merge
GITHUB_REF_PROTECTED=false
---
failures:

---- [codegen] tests/codegen/trailing_zeros.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-15/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/trailing_zeros/trailing_zeros.ll" "/checkout/tests/codegen/trailing_zeros.rs" "--allow-unused-prefixes" "--check-prefixes" "CHECK,NONMSVC" "--dump-input-context" "100"
--- stderr -------------------------------
Build completed unsuccessfully in 0:11:13
/checkout/tests/codegen/trailing_zeros.rs:8:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/trailing_zeros.rs:8:12: error: CHECK: expected string not found in input
 // CHECK: %[[AND:.*]] = and i64 %val, 7
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/trailing_zeros/trailing_zeros.ll:7:45: note: scanning from here
define noundef zeroext i1 @trailing_zeros_ge(i64 noundef %val) unnamed_addr #0 {
                                            ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/trailing_zeros/trailing_zeros.ll:9:24: note: possible intended match here
 %0 = tail call i64 @llvm.cttz.i64(i64 %val, i1 false), !range !3
/checkout/tests/codegen/trailing_zeros.rs:18:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/trailing_zeros.rs:18:12: error: CHECK: expected string not found in input
 // CHECK: %[[ICMP:.*]] = icmp eq i64 [[AND]], 0
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/trailing_zeros/trailing_zeros.ll:18:23: note: scanning from here
 %0 = and i64 %val, 15
                      ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/trailing_zeros/trailing_zeros.ll:18:23: note: with "AND" equal to "0"
 %0 = and i64 %val, 15
                      ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/trailing_zeros/trailing_zeros.ll:19:2: note: possible intended match here
 %_0 = icmp eq i64 %0, 0

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/trailing_zeros/trailing_zeros.ll
Check file: /checkout/tests/codegen/trailing_zeros.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
            1: ; ModuleID = 'trailing_zeros.9e65c6efcd1e645-cgu.0' 
            2: source_filename = "trailing_zeros.9e65c6efcd1e645-cgu.0" 
            3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128" 
            4: target triple = "x86_64-unknown-linux-gnu" 
            5:  
            6: ; Function Attrs: mustprogress nofree nosync nounwind nonlazybind readnone willreturn uwtable 
            7: define noundef zeroext i1 @trailing_zeros_ge(i64 noundef %val) unnamed_addr #0 { 
check:8'0                                                  X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
            8: start: 
check:8'0      ~~~~~~~
            9:  %0 = tail call i64 @llvm.cttz.i64(i64 %val, i1 false), !range !3 
check:8'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:8'1                             ?                                           possible intended match
           10:  %_2 = trunc i64 %0 to i32 
check:8'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~
           11:  %_0 = icmp ugt i32 %_2, 2 
check:8'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~
           12:  ret i1 %_0 
check:8'0      ~~~~~~~~~~~~
           13: } 
check:8'0      ~~
           14:  
check:8'0      ~
           15: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind readnone willreturn uwtable 
check:8'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           16: define noundef zeroext i1 @trailing_zeros_gt(i64 noundef %val) unnamed_addr #1 { 
check:8'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           17: start: 
           18:  %0 = and i64 %val, 15 
check:18'0                           X error: no match found
check:18'1                             with "AND" equal to "0"
           19:  %_0 = icmp eq i64 %0, 0 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
check:18'2      ?                        possible intended match
           20:  ret i1 %_0 
check:18'0     ~~~~~~~~~~~~
           21: } 
check:18'0     ~~
           22:  
check:18'0     ~
           23: ; Function Attrs: mustprogress nocallback nofree nosync nounwind readnone speculatable willreturn 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           24: declare i64 @llvm.cttz.i64(i64, i1 immarg) #2 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           25:  
check:18'0     ~
           26: attributes #0 = { mustprogress nofree nosync nounwind nonlazybind readnone willreturn uwtable "probe-stack"="__rust_probestack" "target-cpu"="x86-64" } 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           27: attributes #1 = { mustprogress nofree norecurse nosync nounwind nonlazybind readnone willreturn uwtable "probe-stack"="__rust_probestack" "target-cpu"="x86-64" } 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           28: attributes #2 = { mustprogress nocallback nofree nosync nounwind readnone speculatable willreturn } 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           29:  
check:18'0     ~
           30: !llvm.module.flags = !{!0, !1} 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           31: !llvm.ident = !{!2} 
check:18'0     ~~~~~~~~~~~~~~~~~~~~
           32:  
check:18'0     ~
           33: !0 = !{i32 7, !"PIC Level", i32 2} 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           34: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           35: !2 = !{!"rustc version 1.73.0-nightly (99a1aac87 2023-08-15)"} 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           36: !3 = !{i64 0, i64 65} 
check:18'0     ~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------



```

</details>

--- a/tests/codegen/trailing_zeros.rs
+++ b/tests/codegen/trailing_zeros.rs
@@ -1,0 +1,22 @@
+// compile-flags: -O
+// min-llvm-version: 17
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @trailing_zeros_ge
+#[no_mangle]
+pub fn trailing_zeros_ge(val: u32) -> bool {
+    // CHECK: %[[AND:.*]] = and i32 %val, 7
+    // CHECK: %[[ICMP:.*]] = icmp eq i32 %[[AND]], 0
+    // CHECK: ret i1 %[[ICMP]]
+    val.trailing_zeros() >= 3
+}
+
+// CHECK-LABEL: @trailing_zeros_gt
+#[no_mangle]
+pub fn trailing_zeros_gt(val: u64) -> bool {
+    // CHECK: %[[AND:.*]] = and i64 %val, 15
+    // CHECK: %[[ICMP:.*]] = icmp eq i64 %[[AND]], 0
+    // CHECK: ret i1 %[[ICMP]]
+    val.trailing_zeros() > 3
+}

